packages/playwright-core/src/server/webkit/wkBrowser.ts

```diff
@@ -256,7 +256,11 @@ export class WKBrowserContext extends BrowserContext {
   async doGetCookies(urls: string[]): Promise<channels.NetworkCookie[]> {
     const { cookies } = await this._browser._browserSession.send('Playwright.getAllCookies', { browserContextId: this._browserContextId });
     return network.filterCookies(cookies.map((c: channels.NetworkCookie) => {
-      const copy: any = { ... c };
+      const copy: any = { ...c };
+      // "Prevent cross-site tracking." is on by default in Safary, which essentially
+      // disables SameSite=None and such cookies become Lax.
+      if (copy.sameSite === 'None')
```

Comment:
I think we should return the cookie as in Set-Cookie header and as in Network stack.